### ISSUE TITLE
jsonpath question

### ISSUE THREAD

**Issue by chenbekor** [josephpconley/play-jsonpath#5]
assuming I have the following json:

val json = Json.parse("""[{"name": "primary_event", "value":"XX"}{"name": "doc_id", "value":"0BykDAwZ35L"}{"name": "doc_value"}]"""")

I want to extract only the value 0BykDAwZ35L , I guess I should use some kind of a filter   -  what is the suggested jsonPath?

**Comment by josephpconley** [comment]
If you want to say "give me the `value` property from the second JSON object then the following would work:

    JSONPath.query("$[1].value", json)

**Comment by chenbekor** [comment]
I don't know the exact index to search  for. your expression assumes that the value will always appear at index 1.i need to somehow filter the array down to that specific item in an idiomatic way.

**Comment by josephpconley** [comment]
So are you trying to find the index based on that value?  How would you
word this query in plain English?

**Comment by chenbekor** [comment]
search in the array a json object with attribute  'name'  equals 'doc_id'  and fetch from that object the 'value'  attribute content.

**Comment by chenbekor** [comment]
@josephpconley here is the json I'm trying to deal with. I need to  process the below event using the following steps:

1. filter jsons with attr name == 'create'
2. filter from #1 only json objects with attr 'name' == 'doc_id'
3 fetch the attr 'value'

Your help is greatly appreciated!

{
  "events": [
    {
      "name": "create",
      "parameters": [
        {
          "boolValue": true,
          "name": "primary_event"
        },
        {
          "name": "owner",
          "value": "x@x.com"
        },
        {
          "name": "doc_id",
          "value": "askjakak"
        },
        {
          "name": "doc_type",
          "value": "folder"
        },
        {
          "name": "doc_title",
          "value": "Aug"
        }
      ],
      "type": "access"
    },
    {
      "name": "add_to_folder",
      "parameters": [
        {
          "boolValue": false,
          "name": "primary_event"
        },
        {
          "multiValue": [
            "2015"
          ],
          "name": "destination_folder_title"
        },
        {
          "multiValue": [
            "akjhsdasjkk"
          ],
          "name": "destination_folder_id"
        },
        {
          "name": "owner",
          "value": "x@x.com"
        },
        {
          "name": "doc_id",
          "value": "askakllkal"
        },
        {
          "name": "doc_type",
          "value": "folder"
        },
        {
          "name": "doc_title",
          "value": "Aug"
        }
      ],
      "type": "access"
    }
  ]
}

**Comment by josephpconley** [comment]
The first step you can accomplish with this:

    $.events[?(@.name == 'create')]

At that point you should get a `JsArray` with all of the matching JSON objects.  I think from there you'd have to do a Scala `filter` on that array to inspect the `parameters` array in each object and return ones with name = doc_id.
